go/worker/keymanager/handler.go

```diff
@@ -3,22 +3,71 @@ package keymanager
 import (
 	"context"
 	"errors"
+	"sync"
 
+	"github.com/oasislabs/oasis-core/go/common/cbor"
+	"github.com/oasislabs/oasis-core/go/keymanager/api"
+	"github.com/oasislabs/oasis-core/go/keymanager/client"
 	"github.com/oasislabs/oasis-core/go/runtime/localstorage"
+	workerCommon "github.com/oasislabs/oasis-core/go/worker/common"
 	"github.com/oasislabs/oasis-core/go/worker/common/host/protocol"
 )
 
 var (
-	errMethodNotSupported = errors.New("worker/keymanager: method not supported")
+	errEndpointNotSupported = errors.New("worker/keymanager: RPC endpoint not supported")
+	errMethodNotSupported   = errors.New("worker/keymanager: method not supported")
 
 	_ protocol.Handler = (*hostHandler)(nil)
 )
 
 type hostHandler struct {
+	sync.Mutex
+
 	w            *Worker
+	remoteClient *client.Client
 	localStorage localstorage.LocalStorage
 }
 
+func (h *hostHandler) initRemoteClient(commonWorker *workerCommon.Worker) {
+	// Wait for the runtime (XXX: Do we need to do this?)
+	if _, err := h.w.runtime.RegistryDescriptor(h.w.ctx); err != nil {
```

Comment:
I don't think this is needed, the key manager client should watch for registry descriptor changes internally.